[Flang][Semantics] Add a semantic check for simd construct

Thirumalai-Shaktivel opened the pull request:
Add:
- Missing semantic check for the SAFELEN clause in the SIMD Order construct

tblah commented on flang/lib/Semantics/check-omp-structure.cpp:
```diff
@@ -2156,6 +2156,16 @@ void OmpStructureChecker::Leave(const parser::OmpClauseList &) {
+
+    // 2.11.5 Simd construct restriction (OpenMP 5.1)
+    if (auto *clause{FindClause(llvm::omp::Clause::OMPC_safelen)}) {
+      if (FindClause(llvm::omp::Clause::OMPC_order)) {
```
Presumably future standards might specify ordering other than concurrent. It would be best to check that here because it would be easy to forget it later.

Thirumalai-Shaktivel replied:
Done!

kiranchandramohan commented:
> Missing tests

Can you move these to a separate NFC patch?

Thirumalai-Shaktivel committed:
[Flang][Semantics] Add a semantic check for simd construct,

Fix: Add a missing semantic check for the SAFELEN clause in the SIMD Order construct

Thirumalai-Shaktivel committed:
[Test] Add a test

Thirumalai-Shaktivel commented:
@kiranchandramohan, Done! I have created a new PR here: https://github.com/llvm/llvm-project/pull/110468

tblah reviewed (approved):
LGTM thanks for the update

kiranchandramohan reviewed (approved):
LG.

NimishMishra reviewed (approved):
LGTM

Thirumalai-Shaktivel commented:
Thanks for the reviews. As this PR is approved, I'm merging it.
